src/test/java/spoon/test/filters/FilterTest.java

```diff
@@ -613,21 +623,8 @@ public void testElementMapFunction() throws Exception {
 		assertEquals(cls.getParent(), cls.map((CtClass<?> c)->c.getParent()).list().get(0));
 	}
 	@Test
-	public void testElementMapConsumableFunction() throws Exception {
-		final Launcher launcher = new Launcher();
-		launcher.setArgs(new String[] {"--output-type", "nooutput","--level","info" });
-		launcher.addInputResource("./src/test/java/spoon/test/filters/testclasses");
-		launcher.run();
-		
-		CtClass<?> cls = launcher.getFactory().Class().get(Tacos.class);
-		cls.map((CtClass<?> c, CtConsumer<Object> out)->out.accept(c.getParent()))
-			.forEach((CtElement e)->{
-				assertEquals(cls.getParent(), e);
-			});
-		assertEquals(cls.getParent(), cls.map((CtClass<?> c, CtConsumer<Object> out)->out.accept(c.getParent())).list().get(0));
-	}
-	@Test
 	public void testElementMapFunctionArray() throws Exception {
+		// contract: when a function returns an array, all non-null values are sent to the next step
```

Comment:
I suggest to add another contract here
//contract2: when input is null then mapping function is not called at all.